src/api-service/__app__/agent_events/__init__.py

```diff
@@ -82,6 +88,16 @@ def on_state_update(
                             state=NodeTaskState.setting_up,
                         )
                         node_task.save()
+            elif state == NodeState.done:
+                # Model-validated.
+                done_data = cast(NodeDoneEventData, state_update.data)
```

Comment:
This should also be guarded on `state_update.data`, since we let it be optional.

Reply:
Fixed.